doc/iterations/2025.4/encode_workbench_state_in_url.adoc

```diff
@@ -0,0 +1,109 @@
+
+== Key Result
+
+The URL should be synchronized with the full state of the workbench.
+
+=== Acceptance Criteria
+
+* The URL is synchronized with the full state of the workbench, which includes:
+** The active view on the left (and its configuration)
+** The opened editors in the center
+** The active view on the right (and its configuration)
+** The focused part of the workbench (one of the active views or the active editor)
+** The selection
+* When resolving a URL that specifies a workbench state, the matching workbench state is set up.
+
+== Solution
+
+* We will rework the URL to change based on the state of the workbench.
+* The base URL for an open project will be `/projects/:projectId/edit`.
+* The state of the workbench will be encoded in the URL through the following parameters:
+** `views`: will encode which views are active, e.g. `explorer%2Cdetails`.
+** `editors`: will encode which editors are opened, through the ID of their input, e.g. `representationId1%2CrepresentationId2%2CrepresentationId3`. (Note: what's the real difference between views and editors in this context? Maybe we should merge views and editors as parts).
+** `configurations`: will encode the configurations of the active views and editors, e.g. `explorer_state_closed%2Cexplorer_tree_activeFilterId_filterId1`.
+** `selection`: will encode the workbench selection, just like it already does, e.g. `id1%2Cid2`.
+** `active`: will encode the active part, e.g. `representationId2`.  
+
+=== Examples
+
+image::images/add_workbench_views_to_url_example1.png[Example 1]
+
+URL: `/projects/projectId/edit/views=explorer%2Cdetails`
+
+* In parameter `views`, `explorer` and `details` are used to encode that both the `Explorer` and `Details` view are active in the left and right panels.
+* Since each part may encode its configuration, and the default state for `Explorer` and `Details` is to be opened, we do not need to encode it into the URL.
+* There are no editors opened, so there is no value for parameter `editors`.
+* There is no selection in the workbench, so there is no value for parameter `selection`.
+* There is no active part in the workbench, so there is no value for parameter `active`.
+
+image::images/add_workbench_views_to_url_example2.png[Example 2]
+
+URL: `/projects/projectId/edit?views=explorer%2Cdetails&configurations=explorer_state_closed%2Cdetails_state_closed`
+
+* In parameter `configurations`:
+** `explorer_state_closed` encodes that the `Explorer` view is closed.
+** `details_state_closed` encodes that the `Details` view is closed.
+
+image::images/add_workbench_views_to_url_example3.png[Example 3]
+
+URL: `/projects/projectId/edit/views=explorer%2CrelatedElements&editors=topographyId%2Ctopography2Id&configurations=topographyId_zoomLevel_75%2CtopographyId_position_123_456&relatedElements_incoming%2CrelatedElements_current%2CrelatedElements_outgoing&selection=id1%2Cid2&active=topographyId`
```

Comment:
See this part of the shape is almost unreadable, by encoding everything using one JSON object as a parameter, we can talk about it more efficiently. Since the system has to be extensible, it has to be understandable